server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java

```diff
@@ -147,60 +129,22 @@ public Settings getSettings() {
         return this.settings;
     }
 
-    /**
-     * Operating System level information.
-     */
-    @Nullable
-    public OsInfo getOs() {
-        return this.os;
-    }
-
-    /**
-     * Process level information.
-     */
-    @Nullable
-    public ProcessInfo getProcess() {
-        return process;
-    }
-
-    /**
-     * JVM level information.
-     */
-    @Nullable
-    public JvmInfo getJvm() {
-        return jvm;
-    }
-
-    @Nullable
-    public ThreadPoolInfo getThreadPool() {
-        return this.threadPool;
-    }
-
-    @Nullable
-    public TransportInfo getTransport() {
-        return transport;
-    }
-
-    @Nullable
-    public HttpInfo getHttp() {
-        return http;
-    }
-
-    @Nullable
-    public PluginsAndModules getPlugins() {
-        return this.plugins;
-    }
-
-    @Nullable
-    public IngestInfo getIngest() {
-        return ingest;
+    // TODO: javadoc on the utility of a method that hides casting
```

Comment:
Can this java doc be added before committing?

Reply:
Javadoc added!